src/CraftingRecipes.cpp

```diff
@@ -283,6 +283,80 @@ cCraftingRecipes::~cCraftingRecipes()
 
 
 
+bool cCraftingRecipes::isNewCraftableRecipe(const cRecipe * a_Recipe, const cItem & a_Item, const std::set<cItem, cItem::sItemCompare> & a_KnownItems)
+{
+	bool ContainsNewItem = false;
+	for (auto Ingredient : a_Recipe->m_Ingredients)
+	{
+		if (
+			(Ingredient.m_Item.m_ItemType == a_Item.m_ItemType) &&
+			(
+				(Ingredient.m_Item.m_ItemDamage == a_Item.m_ItemDamage) ||
+				(Ingredient.m_Item.m_ItemDamage == -1)
+			)
+		)
+		{
+			ContainsNewItem = true;
+		}
+		if (a_KnownItems.find(Ingredient.m_Item) == a_KnownItems.end())
+		{
+			return false;
+		}
+	}
+	return ContainsNewItem;
+}
+
+
+
+
+
+std::vector<UInt32> cCraftingRecipes::findNewRecipesForItem(const cItem & a_Item, const std::set<cItem, cItem::sItemCompare> & a_KnownItems)
+{
+	std::vector<UInt32> Recipes;
+	for (UInt32 i = 0; i<m_Recipes.size(); i++)
+	{
+		if (m_Recipes[i]->m_RecipeName.empty())
+		{
+			continue;
+		}
+		if (isNewCraftableRecipe(m_Recipes[i], a_Item, a_KnownItems))
+		{
+			Recipes.push_back(i + 1);
+		}
+	}
+	return Recipes;
+}
+
+
+
+
+
+std::map<AString, UInt32> cCraftingRecipes::getRecipeNameMap()
```

Comment:
Can we build this map once and reuse it?

Reply:
I guess this is a rhetorical question :-)